tests/unit/dccs/test_aws_iot.py

```diff
@@ -0,0 +1,615 @@
+#                                                                             #
+#  - Redistributions of source code must retain the above copyright notice,   #
+#      this list of conditions and the following disclaimer.                  #
+#                                                                             #
+#  - Redistributions in binary form must reproduce the above copyright        #
+#      notice, this list of conditions and the following disclaimer in the    #
+#      documentation and/or other materials provided with the distribution.   #
+#                                                                             #
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"#
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE  #
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE #
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE  #
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR        #
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF       #
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS   #
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN    #
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)    #
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF     #
+#  THE POSSIBILITY OF SUCH DAMAGE.                                            #
+# ----------------------------------------------------------------------------#
+
+import unittest
+import os
+import json
+import pint
+
+from liota.lib.utilities.si_unit import parse_unit
+from liota.dccs.aws_iot import AWSIoT
+from liota.dcc_comms.mqtt_dcc_comms import MqttDccComms
+from liota.entities.metrics.metric import Metric
+from liota.entities.devices.simulated_device import SimulatedDevice
+from liota.entities.edge_systems.dell5k_edge_system import Dell5KEdgeSystem
+from liota.entities.registered_entity import RegisteredEntity
+from liota.entities.metrics.registered_metric import RegisteredMetric
+from liota.lib.utilities.utility import getUTCmillis
+
+# AWS configurations
+config = {}
+execfile(os.path.dirname(os.path.abspath(__file__)) + '/conf/testAWSIoTProp.conf', config)
+connect_rc = 0
+disconnect_rc = 0
+
+# Create a pint unit registry
+ureg = pint.UnitRegistry()
+
+
+# Monkey patched init method of
+def mocked_init_mqtt_dcc_comms(self, *args, **kwargs):
+    pass
+
+
+# Sampling function
+def sampling_function():
+    pass
+
+
+class MQTTTest(unittest.TestCase):
+    """
+    AWSIoT unit test cases
+    """
+
+    def setUp(self):
+        """
+        Method to initialise the AWSIoT parameters.
+        :return: None
+        """
+
+        # EdgeSystem name
+        self.edge_system = Dell5KEdgeSystem(config['EdgeSystemName'])
+
+    def dict_equal(self, d1, d2, msg="Invalid dict, please check the implementation"):
```

Comment:
Remove this method and replace it with this as an utility function outside the testcase class:
```
def ordered(obj):
    if isinstance(obj, dict):
        return sorted((k, ordered(v)) for k, v in obj.items())
    if isinstance(obj, list):
        return sorted(ordered(x) for x in obj)
    else:
        return obj
```
Reference 1st answer: http://stackoverflow.com/questions/25851183/how-to-compare-two-json-objects-with-the-same-elements-in-a-different-order-equa